pkg/query-service/interfaces/interface.go

```diff
@@ -71,6 +71,8 @@ type Reader interface {
 	GetListResultV3(ctx context.Context, query string) ([]*v3.Row, error)
 	LiveTailLogsV3(ctx context.Context, query string, timestampStart uint64, idStart string, client *v3.LogsLiveTailClient)
 
+	GetDashboardsInfo(ctx context.Context) (*model.DashboardsInfo, error)
+	GetAlertsInfo(ctx context.Context) (*model.AlertsInfo, error)
```

Comment:
Reader interface shouldn't be bothered about this unrelated telemetry detail.

Reply:
Similar telemetry functions are already part of reader so this will require major refactoring

Reply:
Can take it up separately: https://github.com/SigNoz/engineering-pod/issues/1141